### PR TITLE
Refactor url handling

### PR DESCRIPTION
**Pull request by jinxka**
I'm trying to normalize the URLs handling because I noticed that like 'URL' before the endpoints arguments does not accept an empty path.

I created a new method 'set_url_path' which add '/' to the url if no path has been set.

I also modified the is_valid_endpoint to is_valid_url which is a method that verify the format of an url.

### CODE REVIEW

File: wapitiCore/main/wapiti.py [hunk 1]
```diff
@@ -648,16 +648,25 @@ def have_attacks_started(self) -> bool:
...
+def is_valid_url(url_type, url):
+    """Verify if the url provided has the right format"""
     try:
         parts = urlparse(url)
     except ValueError:
+        print('ValueError')
         return False
     else:
         if parts.params or parts.query or parts.fragment:
+            print(_("Error: {} must not contain params, query or fragment!").format(url_type))
```
Comment (devl00p) [inline wapitiCore/main/wapiti.py:665]: This is not good for the URL given to wapiti using -u. You can provide an URL containing parameters if you are using the "url" scope

File: wapitiCore/main/wapiti.py [hunk 2]
```diff
@@ -1034,15 +1043,9 @@ def wapiti_main():
...
+    url = set_url_path(args.base_url)
 
-    parts = urlparse(url)
-    # Set path to '/' if path is Null
-    if not parts.path:
-        url += '/'
-
-    if not parts.scheme or not parts.netloc:
-        print(_("Invalid base URL was specified, please give a complete URL with protocol scheme."))
+    if not is_valid_url('URL', url):
```
Comment (devl00p) [inline wapitiCore/main/wapiti.py:1048]: see first comment

File: wapitiCore/main/wapiti.py [hunk 3]
```diff
@@ -1034,15 +1043,9 @@ def wapiti_main():
             print('')
         sys.exit()
 
-    url = args.base_url
+    url = set_url_path(args.base_url)
```
Comment (devl00p) [inline wapitiCore/main/wapiti.py:1046]: what about  `fix_url_path` ?

**Review by devl00p — CHANGES_REQUESTED** [review]

File: wapitiCore/main/wapiti.py (hunk 1 above)
Comment (jinxka) [reply wapitiCore/main/wapiti.py:665]: I reverted the changes for the main *url*

**Pull request merged**